### PR TITLE
Fix strands getting blocked when consuming values concurrently under high load 

### PR DESCRIPTION
**Pull request by Nuvindu**
## Purpose

Fixes: 
https://github.com/ballerina-platform/ballerina-library/issues/6812
https://github.com/ballerina-platform/ballerina-library/issues/6780

### CODE REVIEW

File: native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Class to hold the static cachedThreadPool for the Pipe package.
+ */
+public class WorkerThreadPool {
+    private static final String THREAD_NAME = "bal-pipe-thread";
+
+    private WorkerThreadPool() {
+    }
+
+    public static final ExecutorService PIPE_EXECUTOR_SERVICE = new ThreadPoolExecutor(0, 500,
```
Comment (ThisaruGuruge) [inline native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java]: This max pool number should be configurable, and 500 is a big number for the general case, IMO.

Comment (ayeshLK) [inline native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java]: Lets move the executor to the `Pipe` class itself as it is used there. And mark it as `private`.

File: native/src/main/java/io/xlibb/pipe/Pipe.java [hunk 2]
```diff
@@ -115,7 +116,7 @@ public static BError produce(Environment env, BObject pipe, Object event, BDecim
             Pipe nativePipe = (Pipe) handle.getValue();
```
Comment (ayeshLK) [inline native/src/main/java/io/xlibb/pipe/Pipe.java]: Um bit confused here, why are we using a Ballerina `handle` at the native level to retrieve a native object ?

File: native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java (hunk 1 above)
Comment (ayeshLK) [reply native/src/main/java/io/xlibb/pipe/thread/WorkerThreadPool.java]: IMO we could have used Java Cached threadpool executor [1] in this case 

[1] - https://www.baeldung.com/java-executors-cached-fixed-threadpool

**Review by xlight05 — APPROVED** [review]

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**